web/components/visibility-observer.tsx

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from 'react'
+
+export function VisibilityObserver(props: {
+  className?: string
+  onVisibilityUpdated: (visible: boolean) => void
+}) {
+  const { className, onVisibilityUpdated } = props
```

Comment:
One way to make this slightly more robust is to do
```const onVisibilityUpdated = useEvent(props.onVisibilityUpdated)```

Then it doesn't matter if the caller uses useCallback, the reference to onVisibilityUpdated won't change.

Reply:
OK, I'll defer to you on what you think good style is with regard to this stuff.